web-src/src/components/PromptSessionPanel.js

```diff
@@ -51,9 +53,13 @@ export function PromptSessionPanel() {
         }}>
           <PromptResultListView style={{ height: '100%' }} onClick={() => setIsOpenPromptEditor(false)} />
         </Flex>
-        <PromptEditor isOpen={isOpenPromptEditor} onClose={() => setIsOpenPromptEditor(false)} />
+        <PromptEditor
+          isOpen={isOpenPromptEditor}
+          onClose={() => setIsOpenPromptEditor(false)}
+          error={promptEditorError}
```

Comment:
From my understanding of the app's architecture, the `PromptSessionPanel` component might be considered a `form` component. It could have its own validation state, which other child components might call to notify about any validation errors. Child components like `PromptSessionSideView` and `PromptEditor` might have their own _internal_ validation states to track their validation logic.

```
PromptSessionPanel (a form element) - has an internal state to track form validation and form controls
  |-- PromptSessionSideView (an input element)
  |-- PromptEditor (an input element) - has its own internal state to track its validation logic and notifies PromptSessionPanel in case of errors
```
- once `PromptEditor` encounters a validation error, it should notify `PromptSessionPanel`
- in turn, `PromptSessionPanel` should notify `PromptSessionSideView` to disable `GenerateButton`

In short, passing `promptEditorError` isn't needed. `PromptEditor` should have its own state to control its validation logic and only notify `PromptSessionPanel` about errors by calling `PromptSessionPanel`'s `setPromptEditorError` method. However, we need to rename it to something like `setPromptValidationError` since other child components that we could have might also fail their own validation logic.

However, I need input from @askayastha22 as well.

Reply:
@slitviachenko 
- From my point of view, even though PromptSessionPanel looks like a form element, it's dynamically created via prompt syntax. So, validation logic for each text field is hard to determine beforehand.
- As for your point about `PromptEditor` having its own state to control validation logic, it is a good catch and we may refactor the code.